### PR TITLE
add pack extension --help

### LINKED ISSUE #1470: pack should support `pack extension *` commands
`pack extension --help` should show (analogously to `pack buildpack --help`):

```
  inspect     Show information about an extension
  package     Package an extension in OCI format. <- made a separate issue, please see https://github.com/buildpacks/pack/issues/1489
  new         Creates basic scaffolding of an extension.
  pull        Pull an extension from a registry and store it locally
  register    Register an extension to a registry
  yank        Yank an extension from a registry
```

We should probably make a separate issue for each one of these commands, as they could be implemented separately.

Edit: Made a separate issue (https://github.com/buildpacks/pack/issues/1489) for `pack extension package`.

### PR DESCRIPTION
**Pull request by itsdarshankumar**
## Summary
Added command to show output for "pack extension --help"

## Output
image

image

#### Before
No such command exists before
image

## Related

Resolves #1470

### CODE REVIEW

**Comment by natalieparellano** [comment]
Apologies @itsdarshankumar, I may be confused - is this PR intended to resolve https://github.com/buildpacks/pack/issues/1606, or just the `--help` aspect of `pack extension`? I think either approach is fine but just want to make sure we've linked to the right issue :)

**Comment by itsdarshankumar** [comment]
> Apologies @itsdarshankumar, I may be confused - is this PR intended to resolve #1606, or just the `--help` aspect of `pack extension`? I think either approach is fine but just want to make sure we've linked to the right issue :)

@natalieparellano This PR is for --help, I'll add another PR for #1606

**Review by jjbustamante — APPROVED** [review]
I think this PR is ok to be merged, Darshan is also working in the implementation of each of the commands for extension on pack, and this one is required. 

CC @samj1912 ,  @hone , @dfreilich , @jkutner

File: internal/commands/extension_package.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	Path              string
+}
+
+// Packager and PackageConfigReader to be added here and argument also to be added in the function
+
+// ExtensionPackage packages (a) extension(s) into OCI format, based on a package config
+func ExtensionPackage(logger logging.Logger, cfg config.Config) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "package <name> --config <config-path>",
+		Short: "Package an extension in OCI format.",
```
Comment (natalieparellano) [inline internal/commands/extension_package.go:26]:
```suggestion
		Short: "Package an extension in OCI format",
```
(nit, to make it more consistent with other commands... `pack buildpack` also has this inconsistency, but we could at least make it better here)

File: internal/commands/extension_new.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	Version string
+}
+
+// extensioncreator type to be added here and argument also to be added in the function
+
+// ExtensionNew generates the scaffolding of an extension
+func ExtensionNew(logger logging.Logger) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "new <id>",
+		Short:   "Creates basic scaffolding of an extension.",
```
Comment (natalieparellano) [inline internal/commands/extension_new.go:23]:
```suggestion
		Short:   "Creates basic scaffolding of an extension",
```

**Review by natalieparellano — APPROVED** [review]

**Comment by natalieparellano** [comment]
@buildpacks/toc any thoughts on this one?

**Review by jkutner — APPROVED** [review]

**Pull request merged**